core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/CoreTransferExtension.java

```diff
@@ -92,16 +90,16 @@ public void initialize(ServiceExtensionContext context) {
 
         var vault = context.getService(Vault.class);
 
-        provisionManager = new ProvisionManagerImpl();
+        var provisionManager = new ProvisionManagerImpl(monitor);
         context.registerService(ProvisionManager.class, provisionManager);
 
         var waitStrategy = context.hasService(TransferWaitStrategy.class) ? context.getService(TransferWaitStrategy.class) : new ExponentialWaitStrategy(DEFAULT_ITERATION_WAIT);
 
         var dataProxyRegistry = new DataProxyManagerImpl();
         context.registerService(DataProxyManager.class, dataProxyRegistry);
 
-
-        asyncMgr = AsyncTransferProcessManager.Builder.newInstance()
+        transferProcessStore = context.getService(TransferProcessStore.class);
```

Comment:
This will likely have to go away since it's going to be set using @Inject

Reply:
Done